src/core/providers/gdal/qgsgdalprovider.cpp

```diff
@@ -1835,12 +1835,30 @@ QList<QgsProviderSublayerDetails> QgsGdalProvider::sublayerDetails( GDALDatasetH
         }
         else
         {
+
+          // Check if the layer has TIFFTAG_DOCUMENTNAME associated with it. If so, use that name.
+          GDALDatasetH datasetHandle = GDALOpen( name, GA_ReadOnly );
+          QString tagTIFFDocumentName = GDALGetMetadataItem( datasetHandle, "TIFFTAG_DOCUMENTNAME", nullptr );
+          if ( tagTIFFDocumentName != nullptr )
```

Comment:
tagTIFFDocumentName is a QString, not a pointer, so can' be nullptr:

```suggestion
          if ( !tagTIFFDocumentName.isEmpty() )
```